src/main/java/mpern/sap/commerce/build/extensioninfo/ExtensionInfoLoader.java

```diff
@@ -0,0 +1,204 @@
+package mpern.sap.commerce.build.extensioninfo;
+
+import static mpern.sap.commerce.build.HybrisPlugin.*;
+
+import java.io.File;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import org.gradle.api.Project;
+import org.gradle.api.file.FileTree;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.tasks.util.PatternSet;
+
+import mpern.sap.commerce.build.HybrisPluginExtension;
+import mpern.sap.commerce.build.util.Extension;
+import mpern.sap.commerce.build.util.ExtensionType;
+import mpern.sap.commerce.build.util.Stopwatch;
+
+/**
+ * Responsible for the creation of
+ * {@link mpern.sap.commerce.build.util.Extension} DTOs for the core build
+ * plugin.
+ */
+public class ExtensionInfoLoader {
+
+    private static final String CUSTOM_DIR = "custom";
+
+    private static final Set<String> PLATFORM_INNER_EXT_NAMES = new HashSet<>(
```

Comment:
Even though they are stable, should we read them from the platform folder itself?

Reply:
It is a good idea, not sure it is worth the effort now. We can add an improvement issue and do it when I have some more time.